### DISCUSSION TITLE
Signin session exposed between browser windows of 2 different application with different set of `NEXTAUTH_<foo>` values

### DISCUSSION THREAD

**Discussion by khteh** [nextauthjs/next-auth#7285] [Help]
### Environment

System:
OS: Linux 5.19 Ubuntu 22.10 22.10 (Kinetic Kudu)
CPU: (16) x64 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz
Memory: 59.74 GB / 70.39 GB
Container: Yes
Shell: 5.2.2 - /bin/bash
Binaries:
Node: 19.8.1 - /usr/local/bin/node
Yarn: 1.22.19 - ~/.yarn/bin/yarn
npm: 9.6.4 - /usr/local/bin/npm
Browsers:
Chrome: 112.0.5615.121
Firefox: 112.0

### Reproduction URL

https://github.com/nextauthjs/next-auth-example

### Describe the issue

Build and start `next-auth-example-1`, signin successfully with a credential.
Build and start `next-auth-example-2` with different set of `NEXTAUTH_<foo>` values. Try to sign in in a separate browser window and it will show `"signed in as"` the credential of the user signed in `next-auth-example-1`.

https://github.com/nextauthjs/next-auth/issues/7280

### How to reproduce

Build and start `next-auth-example-1`, signin successfully with a credential.
Build and start `next-auth-example-2` with different set of `NEXTAUTH_<foo>` values. Try to sign in in a separate browser window and it will show `"signed in as"` the credential of the user signed in `next-auth-example-1`.

### Expected behavior

The second signin should start afresh, redirecting to the provider for the user to input credentials instead of showing being "signed in"

**Comment by balazsorban44** [comment]
Since you are on `localhost`, it might still share cookies as it's the same domain so the second app would still see the first one's session cookie. setting a different `NEXTAUTH_SECRET` for the second app though should invalidate the signed-in cookie, as the mismatching secret would make the session invalid. You can see this logic being applied here: 

https://github.com/nextauthjs/next-auth/blob/e97b27414a5f7a4ca5192ed5ab014df549758d96/packages/next-auth/src/core/routes/session.ts#L163-L167

It would be worth testing this deployed where the domains are fully different to verify it's not an actual `next-auth` bug, but a local setup issue.
